src/pudl_archiver/depositors/fsspec.py

```diff
@@ -72,26 +73,37 @@ def from_upath(cls, deposition_path: UPath):
 class FsspecAPIClient(DepositorAPIClient):
     """Implement API for fsspec based depositors."""
 
-    base_path: str
+    path: str
 
     @classmethod
     async def initialize_client(
         cls,
         session: aiohttp.ClientSession,
         sandbox: bool,
+        deposition_path: str,
     ) -> "FsspecAPIClient":
         """Return initialized fsspec api client."""
-        return cls()
+        logger.warning(
+            "The fsspec depositor backend is in an early/experimental state. "
+            "It currently does not support versioning so any existing archive would be overwritten. "
+            "Please use with caution."
+        )
+        if sandbox:
+            raise NotImplementedError(
+                "There is no sandbox available for fsspec archiver. "
+                "An alternative for testing would be to use the 'local' depositor backend."
+            )
+        return cls(path=deposition_path)
 
     async def get_deposition(self, dataset_id: str):
         """Get latest version of deposition associated with dataset_id."""
         return Deposition.from_upath(
-            deposition_path=UPath(self.base_path) / dataset_id,
+            deposition_path=UPath(self.path),
```

Comment:
Yeah that was my idea. We could definitely change back if that'd be preferred, but I thought it might be confusing if you pass a path and it ends up creating an unexpected sub-directory.

Reply:
Oh no, that seems totally justifiable on second thought!